Fw/DataStructures/RedBlackTreeSetOrMapImpl.hpp

```diff
@@ -0,0 +1,1699 @@
+                    //                             /          \
+                    //                            /            \
+                    //                           /              \
+                    //                          V                V
+                    //             BBBBBBBBBBBBBB                BBBBBBBBBBBBBBBBBBB
+                    //            B              B              B                   B
+                    //            B      K1      B              B    K6 (parent)    B
+                    //            B              B              B                   B
+                    //             BBBBBBBBBBBBBB                BBBBBBBBBBBBBBBBBBB
+                    //                 |    |                        /         \
+                    //                 |    |                       /           \
+                    //                 V    V                      V             V
+                    //   ------------------------      RRRRRRRRRRRRRRRRRR      ------------------------
+                    //   |                      |     R                  R     |                      |
+                    //   |  black height i + 1  |     R   K4 (sibling)   R     |  black height i + 1  |
+                    //   |                      |     R                  R     |                      |
+                    //   ------------------------      RRRRRRRRRRRRRRRRRR      ------------------------
+                    //                                      /      \
+                    //                                     /        \
+                    //                                    V          V
+                    //            BBBBBBBBBBBBBBBBBBBBBBBBBB        BBBBBBBBBBBBBBBBBBBBBBBB
+                    //           B                          B      B                        B
+                    //           B    K3 (distantNephew)    B      B    K5 (closeNephew)    B
+                    //           B                          B      B                        B
+                    //            BBBBBBBBBBBBBBBBBBBBBBBBBB        BBBBBBBBBBBBBBBBBBBBBBBB
+                    //                     |      |                         |      |
+                    //                     |      |                         |      |
+                    //                     V      V                         V      V
+                    //             ------------------------         ------------------------
+                    //             |                      |         |                      |
+                    //             |    black height i    |         |    black height i    |
+                    //             |                      |         |                      |
+                    //             ------------------------         ------------------------
+                    */
+                }
+                done = true;
```

Comment:
Don't we want a break here?

Reply:
Technically, the break is at the end, but still with all the comments done && break seems clearer.